File: src/main/java/org/apache/prepbuddy/utils/Range.java
package org.apache.prepbuddy.utils;

import java.util.ArrayList;
import java.util.List;

public class Range {

    private final double lowerBound;
    private final double uppperBound;

    public Range() {
        this(0, 0);
    }

    public Range(double lowerBound, double uppperBound) {
        this.lowerBound = lowerBound;
        this.uppperBound = uppperBound;
    }

    public boolean contains(double aValue) {
        return aValue >= lowerBound && aValue <= uppperBound;
    }

    public List<Integer> lowerToUpperValues() {
        List<Integer> continuousValues = new ArrayList<>();
        for (int value = (int) lowerBound; value <= uppperBound; value++)
            continuousValues.add(value);
        return continuousValues;
    }
}

File: src/main/scala/org/apache/datacommons/prepbuddy/utils/Range.scala
package org.apache.datacommons.prepbuddy.utils

class Range(lowerBound: Double = 0, upperBound: Double = 0) {
    def this(lowerBound: Int, upperBound: Int) = this(lowerBound.toDouble, upperBound.toDouble)

    def contains(aValue: Double): Boolean = {
        aValue >= lowerBound && aValue <= upperBound
    }
}

